src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -332,14 +332,24 @@ private static int VerifyClientCertificate(int preverify_ok, IntPtr x509_ctx_ptr
 
         private static unsafe int AlpnServerSelectCallback(IntPtr ssl, out IntPtr outp, out byte outlen, IntPtr inp, uint inlen, IntPtr arg)
         {
+            outp = IntPtr.Zero;
+            outlen = 0;
+
+            if (arg == IntPtr.Zero)
```

Comment:
In what situation will it be null?  Could you add a comment?

Reply:
I might remove that check now. I have just moved the GCHandle to be inside the SslCTX handle and tied directly to it's lifetime, so now this method should never be called after the handle is freed. I would perhaps change these to debug.asserts "just in case" but will remove the actual check.